tests/test_tokenization_common.py

```diff
@@ -368,11 +369,21 @@ def test_sentencepiece_tokenize_and_convert_tokens_to_string(self):
         # check if converting back to original text works
         reverse_text = tokenizer.convert_tokens_to_string(tokens)
 
-        if self.test_sentencepiece_ignore_case:
-            reverse_text = reverse_text.lower()
+        # All tokenizers pass this test without the below commented out code.
+        # if self.test_sentencepiece_ignore_case:
+        #     reverse_text = reverse_text.lower()
 
         self.assertEqual(reverse_text, text)
 
+        if isinstance(tokenizer, LayoutXLMTokenizer):
+            return
+
+        input_ids = tokenizer(text).input_ids
+        tokens_including_special = tokenizer.convert_ids_to_tokens(input_ids)
+        reverse_text = tokenizer.convert_tokens_to_string(tokens_including_special)
+
+        self.assertEqual(len(tokenizer.tokenize(reverse_text)), len(input_ids))
```

Comment:
If I understood correctly, the initial problem was that the special token was not found in the `convert_tokens_to_string` output while given as individual tokens in the input. So, I think we should refocus the test on this check by testing the method on all the special tokens of each spm tokenizer - which are stored in the `all_special_tokens` attribute.

So I suggest the following changes:
```suggestion
        special_tokens =  tokenizer.all_special_tokens
        special_tokens_string = tokenizer.convert_tokens_to_string(special_tokens)
        for special_token in special_tokens:
            self.assertIn(special_token, special_tokens_string)
```

These changes cause the test to fail now for `ReformerTokenizer`, `BigBirdTokenizer` and `BertGenerationTokenizer` which are also spm tokenizers. 

Don't hesitate to tell me if you ever wrote the previous test with another case in mind! :smile:

Reply:
Great idea! This test is definitely more robust than my length comparison, especially it it managed to catch a few more SPM tokenizers.